woocommerce/woocommerce-ios
# Insert custom error VC into WPAuthenticator, overriding inline error message

Closes #3218 
Ref: p91TBi-3Rn-p2

⚠️ Please note this PR is not against develop, but against the feature branch for UL&S 🙇

⚠️⚠️ Please note also that the coloring of the CTAs (Back button, Help button and others) is implemented in #3253 

## Design
<img src="https://user-images.githubusercontent.com/2722505/100716098-338ed400-33f3-11eb-92ee-4bdf959b3c4a.png" width="350"/>

⚠️⚠️⚠️ Please note there is a limitation that, as far as I can tell, we can not work around: at the moment of presenting the error there is no way to know the site address the user entered, so I removed it from the error message.

## Changes

First a gif of the new behaviour:
<img src="https://user-images.githubusercontent.com/2722505/100716301-781a6f80-33f3-11eb-9cc3-107607045ddb.gif" width="350"/>


Screenshots:
| Light mode | Dark mode |
| ---- | ---- |
| <img src="https://user-images.githubusercontent.com/2722505/100716412-a1d39680-33f3-11eb-91c2-08e78fd0adaa.png" width="350"/> | <img src="https://user-images.githubusercontent.com/2722505/100716552-d3e4f880-33f3-11eb-96d6-8003e8b10af6.png" width="350"/> |

The changes to the code are:
* Added a new view model to represent this error
* Pointed WPAuthenticator to the corresponding branch
* Implemented the new methods declared in WPAuthenticatorDelegate, and passed a properly customized error UI to WPAuthenticator.

## How to test
* Check out the branch. Run `bundle exec pod install`
* Log out if necessary
* Attempt to log in with a store address. Use an address that does not correspond to a WP site (i.e. GitHub.com)
* Note: The "Log In With Another Account" button in the error screen should pop to the Login Prologue. The "Enter Another Site" button is deactivated in order to submit the PR, as the functionality to support t is part of a different PR in WPAuthenticator wordpress-mobile/WordPressAuthenticator-iOS#532 . Making progress in two different repos with the desired level of granularity is hard!

Update release notes:
